src/drive/components/FilesViewer.jsx

```diff
@@ -39,24 +42,35 @@ class FilesViewer extends Component {
       <Viewer
         files={files}
         currentIndex={currentIndex}
-        onChange={nextFile =>
+        onChange={nextFile => {
           router.push({
             pathname: `${getParentPath(router)}/${nextFile.id}`
           })
-        }
-        onClose={() =>
+        }}
+        onClose={() => {
           router.push({
             pathname: getFolderUrl(params.folderId, router.location)
           })
-        }
+        }}
+        fetchMore={() => {
+          this.props.fetchMoreFiles(params.folderId, files.length, LIMIT)
+        }}
```

Comment:
I am not sure the `fetchMore` is under the responsibility of the `Viewer`?

The other solution is to keep the whole code in the `FilesViewer` component, in its `render()` function.

What do you think?

Reply:
Yeah, the `fetchMore` shouldn't be under the resp of the `Viewer`, what I had in mind was to keep the code in the `FilesViewer`, exactly.